src/main/java/frc/robot/subsystems/BatterySubsystem.java

```diff
@@ -0,0 +1,120 @@
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import edu.wpi.first.wpilibj.DriverStation;
+
+/** Add your docs here. */
+// Put methods for controlling this subsystem
+// here. Call these from Commands.
+public class BatterySubsystem extends SubsystemBase {
+
+  private ShuffleboardTab btTab;
+  private NetworkTableEntry sbVoltage;
+  private NetworkTableEntry sbInCurrent;
+  private NetworkTableEntry sbTimer;
+  private NetworkTableEntry sbTimerChange;
+  private NetworkTableEntry sbTimerHighCurrent;
+  private DriverStation driverStation;
+  private Timer timer = new Timer();
+  private Timer currentTimer = new Timer();
+
+  private void init() {
+    //timer.reset(); Only uncomment this if you want the timer to reset each time the robot redeploys code
+    btTab = Shuffleboard.getTab("Battery");
+    sbVoltage = btTab.add("Battery Voltage", 0).withSize(2, 2).withPosition(0, 0).getEntry();
+    sbInCurrent = btTab.add("Input Current", 0).withSize(2, 2).withPosition(2, 0).getEntry();
+    sbTimer = btTab.add("Timer", 0).withSize(2, 2).withPosition(0, 2).getEntry();
+    sbTimerHighCurrent = btTab.add("High Current Timer", 0).withSize(2, 2).withPosition(0, 4).getEntry();
+    sbTimerChange = btTab.add("Change Timer", 0).withSize(2, 2).withPosition(4, 0).getEntry();
+    timer.start();
+    currentTimer.start();
+  }
+
+  public BatterySubsystem() {
+    init();
+  }
+
+  public void periodic() {
+    sbVoltage.setDouble(getVoltage());
+    sbInCurrent.setDouble(getInputCurrent());
+    sbTimer.setDouble(getTimer());
+    sbTimerHighCurrent.setDouble(getTimer());
+    sbTimerChange.setBoolean(
+        checkTimer()); // Replace checkVoltage() with checkTimer() if necessary
+    checkCurrent();
+  }
+
+  public double getVoltage() {
+    return RobotController.getBatteryVoltage();
+  }
+
+  public double getInputCurrent() {
+    return RobotController.getInputCurrent();
+  }
+  public void checkCurrent() {
+    if(getInputCurrent() < Constants.maxBatteryCurrent) {
+      Timer.delay(0.02); //Was current timer, but I got a warning about it needing to be static so I changed it to this. Will this cause conflicts?
+    }
+  }
+
+
+  public double getTimer() {
+    return Timer.getFPGATimestamp();
+  }
+
+  public boolean checkTimer() {
+    if (currentTimer.hasElapsed(Constants.timeInSecondsHighCurrentRed)) {
+      DriverStation.reportError("Change the Battery!", true);
+      return true;
+    } else if (currentTimer.hasElapsed(Constants.timeInSecondsGeneralRed)) {
+      DriverStation.reportError("Change the Battery!", true);
+      return true;
+    } else if (timer.hasElapsed(Constants.timeInSecondsHighCurrentYellow)) {
+      DriverStation.reportError("Change the Battery Soon!", true);
+      return false;
+    } else if (timer.hasElapsed(Constants.timeInSecondsGeneralYellow)) {
+      DriverStation.reportError("Change the Battery Soon!", true);
+      return false;
+    }
+    return false;
```

Comment:
Each time we redeploy code these timers will be reset. This may not true of Timer.getFPGATimestamp(), but that is somewhat unclear. Either way tracking time elapsed across redeploys presents a problem.

Reply:
TBD

Reply:
done, timer values are now saved to the robot with Preferences